stress/stress.ml

```diff
@@ -12,53 +13,83 @@ let job_log job =
          let open L.Write in
          release_param_caps ();
          let msg = Params.msg_get params in
-         Buffer.add_string job msg;
+         Option.iter (fun ch -> output_string ch msg) ch;
          Capnp_rpc_lwt.Service.(return (Response.create_empty ()))
      end
 
 let ocaml_package_name = "ocaml-base-compiler"
 let ocaml_version = "4.14.0"
 let opam_repository_commit = "34576e67c88137d40ce6ff9e252d549e9e87205f"
 
+(* XXX: vars is the same on all platforms! *)
 let package_to_custom vars package =
-  let+ opamfile = Utils.get_opam_file package in
+  let opamfile = Lwt_eio.run_lwt (fun () -> Utils.get_opam_file package) in
   Solver_service_api.Worker.Solve_request.
     {
       opam_repository_commits =
-        [ ("git@github.com:ocaml/opam-repository.git", opam_repository_commit) ];
+        [ ("https://github.com/ocaml/opam-repository.git", opam_repository_commit) ];
       root_pkgs = [ (package, opamfile) ];
       pinned_pkgs = [];
       platforms = [ ("macOS", vars); ("linux", vars); ("windows", vars) ];
     }
 
-let requests log solver =
-  let* vars = Utils.get_vars ~ocaml_package_name ~ocaml_version () in
+let solve solver ~log request =
+  Lwt_eio.run_lwt @@ fun () ->
+  Solver_service_api.Solver.solve solver ~log request
+
+let benchmark ~log solver =
+  let vars = Utils.make_vars ~ocaml_package_name ~ocaml_version () in
   let packages = [ "fmt.0.9.0"; "alcotest.1.6.0"; "yojson.2.0.2" ] in
-  let* requests = Lwt_list.map_p (package_to_custom vars) packages in
+  let requests = Fiber.List.map (package_to_custom vars) packages in
+  (* Warm-up *)
   let before = Unix.gettimeofday () in
-  let* results =
-    Lwt_list.map_p (Solver_service_api.Solver.solve solver ~log) requests
-  in
-  let+ () =
-    Lwt_list.iter_s
-      (fun r ->
-        Lwt.return
-        @@ Yojson.Safe.pp Format.std_formatter
-        @@ Solver_service_api.Worker.Solve_response.to_yojson r)
-      results
-  in
-  before
+  let results = Fiber.List.map (fun req -> req, solve solver ~log req) requests in
+  List.iter
+    (fun (_, r) ->
+       Yojson.Safe.pretty_print Format.std_formatter
+       @@ Solver_service_api.Worker.Solve_response.to_yojson r;
+       match r with
+       | Ok [_; _; _] -> ()
+       | _ -> failwith "Warm-up solve failed!"
+    )
+    results;
+  let time = Unix.gettimeofday () -. before in
+  Format.printf "@.Solved warm-up requests in: %.2fs@." time;
+  let expected = results in
+  (* Main run *)
+  let n_iters = 10 in
+  Format.printf "Running another %d rounds...@." n_iters;
+  let before = Unix.gettimeofday () in
```

Comment:
Eio could also export a monotonic/cputime clock measurement to make this sort of benchmarking more accurate -- we use gettimeofday in a few places in this diff already.